### PR TITLE
LIDO-1677

### PR DESCRIPTION
**Pull request by marhong2**

### CODE REVIEW

File: formulas/functions/statistical.js [hunk 1]
```diff
@@ -328,7 +328,41 @@ const StatisticalFunctions = {
...
+    if(otherCriteria.length > 252)
+      throw FormulaError.TOO_MANY_ARGS;
+
+    const range = H.flattenDeep(H.accept(minRange))
+    const criteria_ranges = []
+    const criteria = []
+
+    otherCriteria.forEach((val, idx) => {
+      const accepted = H.accept(val)
+      if(idx % 2 == 0 && typeof accepted != "object")
```
Comment (tyleradams) [inline formulas/functions/statistical.js:341]: === and !==

File: formulas/functions/statistical.js [hunk 2]
```diff
@@ -328,7 +328,41 @@ const StatisticalFunctions = {
...
+      throw FormulaError.VALUE
+
+    criteria_ranges.forEach((val) => {
+      if(val.length != range.length)
+        throw FormulaError.VALUE
+    })
+    
+    const arr = []
+    range.forEach((val, idx) => {
+       const allTrue = criteria_ranges.reduce((prev, curr, criteriaIdx) => {
```
Comment (tyleradams) [inline formulas/functions/statistical.js:358]: use criteria_ranges.every

**Review by tyleradams — CHANGES_REQUESTED** [review]

File: formulas/functions/statistical.js [hunk 3]
```diff
@@ -328,7 +328,40 @@ const StatisticalFunctions = {
...
+    criteria_ranges.forEach((val) => {
+      if(val.length != range.length)
+        throw FormulaError.VALUE
+    })
+    
+    const arr = []
+    range.forEach((val, idx) => {
+       const allTrue = criteria_ranges.every((prev, curr, criteriaIdx) => {
+         return prev && curr[idx] == criteria[criteriaIdx]
+       }, true)
```
Comment (tyleradams) [inline formulas/functions/statistical.js:359]: This is unneeded every takes in one parameter

**Review by tyleradams — APPROVED** [review]

**Comment by tyleradams** [comment]
@PrateekshaDubey can you verify the test cases are correct?

**Review by tyleradams — APPROVED** [review]

**Comment by PrateekshaDubey** [comment]
@tyleradams I have verified the test cases those are there in the Microsoft support document attached to Linear Ticket (LIDO - 1677), on high level those seems correct, more insight can be provided while executing these on particular functionality.

**Pull request merged**